lsp-fiber/src/rpc.mli

```diff
@@ -1,6 +1,16 @@
 open! Import
 open Jsonrpc
 
+module Reply : sig
+  type 'resp t
+
+  val now : ('r, Jsonrpc.Response.Error.t) result -> 'r t
+
+  val later :
+       ((('r, Jsonrpc.Response.Error.t) result -> unit Fiber.t) -> unit Fiber.t)
+    -> 'r t
```

Comment:
I _think_ I understand `Reply.now`–it's a value that's just returned immediately–but `Reply.later` is a little tough for me to grok. Your local definition of `later` in `ocaml_lsp_server.ml` was a bit enlightening.
```ocaml
Reply.later (fun k ->
    let open Fiber.O in
    let* resp = f state req in
    k resp)
```
I can kind of see that `f` here is "the thing that you want to be done asynchronously", but I'm having trouble pinning down what `k` is and who provides it. I'm guessing that's the secret to how this change makes concurrency possible.

Also `Rpc.Reply` seems similar to `Jsonrpc_fiber.Reply` in its form and handling, I presume it hands off the concurrency concerns down to `Fiber` in a similar way to how `Rpc.Reply` hands them down to `Jsonrpc_fiber.Reply`. Any clarification would be much appreciated 😁

Reply:
> I can kind of see that f here is "the thing that you want to be done asynchronously", but I'm having trouble pinning down what k is and who provides it. I'm guessing that's the secret to how this change makes concurrency possible.

The secret is in jsonrpc_fiber. To summarize, we spawn a handler fiber to serve the current request. We give the handler's  fiber a continuation `k` that it may use to send the response. Concurrently, we go on to serve the next request.

> Also Rpc.Reply seems similar to Jsonrpc_fiber.Reply in its form and handling, I presume it hands off the concurrency concerns down to Fiber in a similar way to how Rpc.Reply hands them down to Jsonrpc_fiber.Reply

That's exactly it. Rpc.Rply is translated to the corresponding Jsonrpc_fiber.Reply, which goes on to serve the requests concurrently. Note that `now` isn't strictly necessary, as it's fine to send all responses with `later`. I might simplify that later.